app/models/submission/CharityPartnerTransformer.scala

```diff
@@ -16,7 +16,65 @@
 
 package models.submission
 
+import java.time.LocalDate
+
+import play.api.libs.functional.syntax._
+import play.api.libs.json.Reads.{JsObjectReducer, _}
+import play.api.libs.json.{__, _}
+
 class CharityPartnerTransformer extends JsonTransformer {
 
-  //TODO for Partner
+  val localPath: JsPath = __ \ 'partner
+  private val action = "1"
+  private val relationOO = "1"
+  private val relationAO = "2"
+  private val partnerTypeIndividual = "1"
+
+  def userAnswersToIndividualDetails : Reads[JsObject] = {
+
+    val nino = (__ \ "officialsNino").read[String].map{
+      value => JsString(value.filterNot(_.isWhitespace))
+    }
+    (
+      getName(__ \ 'individualDetails\ 'name, __ \ 'officialsName) and
```

Comment:
no json transaformation is written for nameChangeReason, nameChangeReasonOther, emailAddress, nationalIdentityNumber, nationalIDCardIssuingCountry and nationalIDCardExpiryDate.
Do we have valid ticket to revisit the json transformation for all missing B journey changes?

Reply:
We don't have ticket to revisit but I've added comment to create new ticket